cupyx/scipy/interpolate/_bspline.py

```diff
@@ -0,0 +1,912 @@
+
+import operator
+from itertools import product
+
+import cupy
+from cupy._core import internal
+from cupy._core._scalar import get_typename
+
+from cupyx.scipy.sparse import csr_matrix
+
+import numpy as np
+
+TYPES = ['float', 'double', 'complex<float>', 'complex<double>']
+INT_TYPES = ['int', 'long long']
```

Comment:
Why long long? (My ignorance is showing, yes.)

Reply:
`long long` maps to `int64` and `int` to `int32`, which can occur here: https://github.com/cupy/cupy/blob/9dac7f88fb0f22d2e29d56f70d36744c43a70a4b/cupyx/scipy/interpolate/_bspline.py#L690

Reply:
(`long` is 32bit integer on Windows)